sfepy/terms/terms_hyperelastic_base.py

```diff
@@ -1,8 +1,79 @@
 import numpy as nm
 from sfepy.terms.terms import Term, terms
+from sfepy.base.base import Struct
+import six
 
 _msg_missing_data = 'missing family data!'
 
+class HyperElasticFamilyData(Struct):
+    """
+    Base class for hyperelastic family data.
+
+    The common (family) data are cached in the evaluate cache of state
+    variable.
+    """
+    data_shapes = {
+            'mtx_f': ('n_el', 'n_qp', 'dim', 'dim'),
+            'det_f': ('n_el', 'n_qp', 1, 1),
+            'sym_b': ('n_el', 'n_qp', 'sym', 1),
+            'tr_b': ('n_el', 'n_qp', 1, 1),
+            'in2_b': ('n_el', 'n_qp', 1, 1),
+            'sym_c' : ('n_el', 'n_qp', 'sym', 1),
+            'tr_c' : ('n_el', 'n_qp', 1, 1),
+            'in2_c' : ('n_el', 'n_qp', 1, 1),
+            'sym_inv_c' : ('n_el', 'n_qp', 'sym', 1),
+            'green_strain': ('n_el', 'n_qp', 'sym', 1),
+            'inv_f': ('n_el', 'n_qp', 'dim', 'dim'),
+    }
+
+    def init_data_struct(self, state_shape, name='family_data'):
+        n_el, n_qp, dim, n_en, n_c = state_shape
+        sym = dim * (dim + 1) / 2
```

Comment:
You can use

```
sym = dim * (dim + 1) // 2; sym
```

to make pyflakes happy.

Also, use the integer division here! It might be worth to replace all sym computations by calls to dim2sym(), so that those errors do not multiply.